spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/ribbon/okhttp/RetryableOkHttpLoadBalancingClient.java

```diff
@@ -134,7 +134,7 @@ public RequestSpecificRetryHandler getRequestSpecificRetryHandler(OkHttpRibbonRe
 		return new RequestSpecificRetryHandler(false, false, RetryHandler.DEFAULT, null);
 	}
 
-	static class RetryPolicy extends FeignRetryPolicy {
+	static class RetryPolicy extends InterceptorRetryPolicy {
```

Comment:
why this change?

Also this is going to get a littler murky in the 2.0.x branch where I made some changes to this due to moving ribbon to its own module

Reply:
I think this was weird since it wasn't specific to feign.

Reply:
It was weird, but I did it to reuse code instead of copying it everywhere.  I didnt want to rename the class either because it was already out in the wild.  I didnt like it to begin with.  Since I was reusing the code from Feign I am nervous that making this change would cause some issues.  But my memory could be completely wrong as well...

Reply:
The InterceptorRetryPolicy is used for this fix because it will choose and set a 'RibbonServer' instance on the LoadBalancerContext if its ServiceInstance is null.  This allows the RibbonLoadBalancedRetryPolicy to use the loadBalancerContext.serviceInstance for updating ServerStats on circuit tripping exceptions.

prior to this fix the LoadBalancerCommand would choose and set the ServiceInstance; however, that instance is not a RibbonServer and it could not be used for obtaining ServerStats from the LoadBalancer.

Reply:
it appears as though changes to ribbon and or commons  have made the InterceptorRetryPolicy no longer required.